app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -1336,6 +1337,19 @@ class BrowserFragment : WebFragment(), LifecycleObserver, View.OnClickListener,
             hideFindInPage()
         }
 
+        fun bringURLTLDIntoView() {
+            if (urlView != null) {
+                val paint = urlView!!.paint
+                val charWidth = paint.measureText("a", 0, 1)
+                val urlViewWidth = urlView!!.measuredWidth
+                val numChars = urlViewWidth / charWidth
+                if (url.length > numChars) {
+                    val index = UrlUtils.getIndexOfTLD(url)
+                    urlView?.bringPointIntoView(index)
```

Comment:
Note that for URLs we load Gecko already does the extraction internally. In Fennec we get the base domain (public suffix + 1) from Gecko whenever something loads. It may be possible to ask the GeckoView team to extend the API that provides the URL.